src/deepmol/unsupervised/base_unsupervised.py

```diff
@@ -83,13 +94,15 @@ class PCA(UnsupervisedLearn):
     """
 
     def __init__(self,
-                 n_components: Union[int, float, str] = None,
+                 n_components: Union[int, float, str] = 2,
                  copy: bool = True,
                  whiten: bool = False,
                  svd_solver: str = 'auto',
                  tol: float = 0.0,
                  iterated_power: Union[int, str] = 'auto',
-                 random_state: int = None):
+                 random_state: int = None,
+                 n_oversamples=10,
+                 power_iteration_normalizer="auto") -> None:
```

Comment:
Isn't this overkilling? adding so many parameters instead of using kwargs? Maybe you have a justification for that, I am just asking

Reply:
Yeah it is not necessary. It was more so the user knows which parameters can pass but I can put the docstrings of each parameter kwargs. Do you think I should change it?

Reply:
Regarding the issue that you mentioned: If we implement those strategies (compute similarities and so on) it would change dataset._X so no problem (unless you wanto to keep both arrays). If is something that is not supported by DeepMol the user can provide those by creating a SmilesDataset object, it is not required to be through a Loader. Am I understanding your point right?

Reply:
> Yeah it is not necessary. It was more so the user knows which parameters can pass but I can put the docstrings of each parameter kwargs. Do you think I should change it?

I think so, just to keep consistency with what we are doing with the rest of the code. I saw somewhere that you documented the kwargs, which I think is adequate.

Reply:
> Regarding the issue that you mentioned: If we implement those strategies (compute similarities and so on) it would change dataset._X so no problem (unless you wanto to keep both arrays). If is something that is not supported by DeepMol the user can provide those by creating a SmilesDataset object, it is not required to be through a Loader. Am I understanding your point right?

Yes, you are, and you are right, I was not expressing myself right though. My point was whether is worth it to implement for BOD or not, because I think it is worth implementing the similarities and setting the dataset._X as the similarity matrix.